redisvl/extensions/session_manager/standard_session.py

```diff
@@ -84,19 +130,36 @@ def drop(self, id_field: Optional[str] = None) -> None:
                 If None then the last entry is deleted.
         """
         if id_field:
-            messages = self._client.lrange(self.key, 0, -1)
-            messages = [json.loads(msg) for msg in messages]
-            messages = [msg for msg in messages if msg["id_field"] != id_field]
-            messages = [json.dumps(msg) for msg in messages]
-            self.clear()
-            self._client.rpush(self.key, *messages)
+            sep = self._index.key_separator
+            key = sep.join([self._index.schema.index.name, id_field])
         else:
-            self._client.rpop(self.key)
+            key = self.get_recent(top_k=1, raw=True)[0]["id"]  # type: ignore
+        self._index.client.delete(key)  # type: ignore
 
     @property
     def messages(self) -> Union[List[str], List[Dict[str, str]]]:
         """Returns the full chat history."""
-        return self.get_recent(top_k=-1)
+        # TODO raw or as_text?
+        return_fields = [
+            self.id_field_name,
+            self.session_field_name,
+            self.user_field_name,
+            self.role_field_name,
+            self.content_field_name,
+            self.tool_field_name,
+            self.timestamp_field_name,
+        ]
+
+        query = FilterQuery(
+            filter_expression=self._tag_filter,
+            return_fields=return_fields,
+        )
+
+        sorted_query = query.query
```

Comment:
We added sort by support to the redisvl query classes -- so you should be able to do this there?